### PR TITLE
User can select specific QR code, view data, edit, and save their changes.

### PR DESCRIPTION
**Pull request by Kiptong**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -41,9 +46,32 @@
...
+          <form>
+            <div class="form-group">
+              <label for="formQrUrl">Would you like to change your QR destination?</label>
+              <input type="text" class="form-control" id="updatedqrurl" placeholder="ex. www.google.com"/>
+              <hr/>
+              <label for="formQrUrl">What would you like to change your QR name?</label>
+              <input type="text" class="form-control" id="updatedqrname" placeholder="ex. coffee mug"/>
+              <hr/>
+              <label for="formQrUrl">Would you like add a description or update your current description?(optional)</label>
+              <input type="text" class="form-control" id="updatedqrdescrip" placeholder="ex. #1 Dad Coffee Mug"/>
```
Comment (victormejia) [inline public/index.html]: fix the ids for these inputs, like `update-qr-description`

File: public/main.js [hunk 2]
```diff
@@ -1,5 +1,38 @@
 const qrButton = document.getElementById('createqr')
 
+window.onload = qreator()
```
Comment (victormejia) [inline public/main.js]: your calling this function when the page loads, you want to call it once `onload` fires:

```js
window.onload = () => {
  qreator()
}
```

File: public/main.js [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+      })
+      if (data.length > 0) {
+        const $cardHeader = document.getElementById('cardheader')
+        $cardHeader.classList.remove('hidden')
+      }
+    })
+    .catch((err) => console.log(err))
+}
+
+function updateCard() {
```
Comment (victormejia) [inline public/main.js]: this function name is a little ambiguous with `updateQRData`. Maybe a better name?

File: public/main.js [hunk 4]
```diff
@@ -8,33 +41,69 @@ function createCard(data) {
...
+    },
+    body: JSON.stringify(qrUpdate)
+  })
     .then(res => res.json())
-    .then(data => {
-      const $cardRow = document.getElementById('cardrow')
-      data.forEach((data) => {
-        $cardRow.appendChild(createCard(data))
+      .then((data) => {
+        updateCard()
```
Comment (victormejia) [inline public/main.js]: indentation is off here

File: index.js [hunk 5]
```diff
@@ -28,6 +28,23 @@ app.post('/generate', (req, res) => {
...
+    qrCode.qr = url
+    const query = knex('qrcodes')
+      .where('id', '=', req.params.id)
+      .update(qrCode)
+      .returning('*')
+
+    query.then((data) => {
+      res.json(data)
+    })
+      .catch((err) => console.log(err))
```
Comment (victormejia) [inline index.js]: fix spacing and indentation here for clarity:

```js
query
  .then((data) => {
  
  })
  .catch(...)
```

File: index.js [hunk 6]
```diff
@@ -28,6 +28,23 @@ app.post('/generate', (req, res) => {
   })
 })
 
+app.put('/qrcards/:id', (req, res) => {
+  QRcode.toDataURL(req.body.url, {scale: 25}, (err, url) => {
```
Comment (victormejia) [inline index.js]: before doing any of this, store `req.body` in a variable, and use that

File: index.js [hunk 7]
```diff
@@ -28,6 +28,23 @@ app.post('/generate', (req, res) => {
...
+    if (err) throw console.error(err)
+    const qrCode = req.body
+    qrCode.qr = url
+    const query = knex('qrcodes')
+      .where('id', '=', req.params.id)
+      .update(qrCode)
+      .returning('*')
```
Comment (victormejia) [inline index.js]: use newlines to separate the different sets of actions here

**Review by victormejia — CHANGES_REQUESTED** [review]

File: index.js (hunk 7 above)
Comment (Kiptong) [inline index.js]: const query = knex('qrcodes')
      .where('id', '=', req.params.id)

      .update(qrCode)

      .returning('*')



Is this what you are suggesting?

**Review by victormejia — APPROVED** [review]

**Pull request merged**